drivers/xen/regions.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2023 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/xen/generic.h>
+
+#include <kernel_arch_interface.h>
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/arch_interface.h>
+#include <zephyr/sys/mem_blocks.h>
+
+LOG_MODULE_REGISTER(xen_regions);
+
+#define EXTENDED_REGIONS_IDX 1
+#define XEN_HYP_NODE DT_INST(0, xen_xen)
+
+static K_MUTEX_DEFINE(ext_regions_lock);
+
+struct extended_regions {
```

Comment:
static?

Reply:
Why? It's type definition.

Reply:
It is not used anywhere except this file. So, we can make type declaration static (at least it has pretty generic name and may break something in future).

Reply:
Type declaration is always local to the translation unit, as it does not produce any symbols (in C at least).

Reply:
@lorc thanks for clarification, sorry for the noise